tests/lib/rules/generator-star-spacing.js

```diff
@@ -404,6 +404,24 @@ ruleTester.run("generator-star-spacing", rule, {
             options: [{ before: false, after: false }]
         },
 
+        // full configurability
+        {
+            code: "function * foo(){}",
+            options: [{ before: false, after: false, named: "both" }]
+        },
+        {
+            code: "var foo = function * (){};",
+            options: [{ before: false, after: false, anonymous: "both" }]
+        },
+        {
+            code: "class Foo { * foo(){} }",
+            options: [{ before: false, after: false, method: "both" }]
+        },
+        {
+            code: "class Foo { static * foo(){} }",
+            options: [{ before: false, after: false, static: "both" }]
+        },
```

Comment:
Can you add tests where an override is used, but that override doesn't apply to the given code?

e.g.

```js
{
    code: "function* foo() {}",
    options: [{ before: false, after: true, method: "before" }]
}
```